test/action_view/component_test.rb

```diff
@@ -354,6 +354,15 @@ def test_conditional_rendering
                     ""
   end
 
+  def test_render_callbacks
+    events = []
+
+    component_instance = CallbacksComponent.new(events: events)
+
+    assert_includes component_instance.render_in(controller.view_context), "Rendered"
+    assert_equal [:before, :before_around, :after, :after_around], events
```

Comment:
passing the events and relying on it being used by reference feels a bit strange. 
```suggestion
    assert_equal [:before, :before_around, :after, :after_around], component_instance.events
```